src/sentry/replays/usecases/query/__init__.py

```diff
@@ -43,6 +43,52 @@
 from sentry.replays.usecases.query.fields import ComputedField, TagField
 from sentry.utils.snuba import raw_snql_query
 
+VIEWED_BY_ME_KEY_ALIASES = ["viewed_by_me", "seen_by_me"]
+NULL_VIEWED_BY_ID_VALUE = 0  # default value in clickhouse
+
+
+def handle_viewed_by_me_filters(
+    search_filters: Sequence[SearchFilter | str | ParenExpression], request_user_id: int | None
+) -> Sequence[SearchFilter | str | ParenExpression]:
+    """Translate "viewed_by_me" as it's not a valid Snuba field, but a convenience alias for the frontend"""
+    new_filters = []
+    for search_filter in search_filters:
+        if (
+            not isinstance(search_filter, SearchFilter)
+            or search_filter.key.name not in VIEWED_BY_ME_KEY_ALIASES
+        ):
+            new_filters.append(search_filter)
+            continue
+
+        # since the value is boolean, negations (!) are not supported
+        if search_filter.operator != "=":
+            raise ParseError(f"Invalid operator specified for `{search_filter.key.name}`")
+
+        value = search_filter.value.value
+        if not isinstance(value, str) or value.lower() not in ["true", "false"]:
+            raise ParseError(f"Could not parse value for `{search_filter.key.name}`")
+        value = value.lower() == "true"
+
+        if request_user_id is None:
```

Comment:
Decided with @cmanallen not going to test this block. It's a very rare edge case that can only be triggered running the `delete_replays` script with viewed_by_me as a tag filter, which we don't expect to happen.

Reply:
The only call site for this function is in `query_using_optimized_search`, and `request_user` is passed down by either 1) the replay index endpoint or 2) delete_replays script. We expect that future callers are exclusively Django endpoints, which automatically return 401 unauthorized for anonymous/not-logged-in users. So _other than the delete script, `request_user_id` will always be non-None._

Reply:
Could add a comment for this?